clients/drcachesim/tracer/output.cpp

```diff
@@ -975,7 +975,10 @@ process_and_output_buffer(void *drcontext, bool skip_size_cap)
         if (op_offline.get_value() && op_split_windows.get_value())
             buf_ptr += instru->append_thread_exit(buf_ptr, dr_get_thread_id(drcontext));
     }
-    if (!skip_size_cap &&
+
+    uintptr_t mode = tracing_mode.load(std::memory_order_acquire);
+
+    if (!skip_size_cap && mode != BBDUP_MODE_L0_FILTER &&
```

Comment:
This means we don't respect that max refs and max size thresholds if we're still filtering. Could you add a comment clarifying this?

Reply:
Added a comment.